std-bits/table/src/main/java/org/enso/table/data/table/Table.java

```diff
@@ -288,16 +285,14 @@ public Table selectColumns(List<String> colNames) {
    *
    * {@code rightColumnsToDrop} allows to drop columns from the right table that are redundant when joining on equality of equally named columns.
    */
-  public Table join(Table right, List<JoinCondition> conditions, boolean keepLeftUnmatched, boolean keepMatched, boolean keepRightUnmatched, boolean includeLeftColumns, boolean includeRightColumns, List<String> rightColumnsToDrop, String right_prefix) {
+  public Table join(Table right, List<JoinCondition> conditions, boolean keepLeftUnmatched, boolean keepMatched, boolean keepRightUnmatched, boolean includeLeftColumns, boolean includeRightColumns, List<String> rightColumnsToDrop, String right_prefix, Comparator<Object> comparator) {
     // TODO adding prefix for right columns
     NameDeduplicator deduplicator = new NameDeduplicator();
 
-    JoinStrategy strategy = new ScanJoin();
-    JoinResult joinResult = null;
+    JoinStrategy strategy = new IndexJoin(comparator);
+
     // Only compute the join if there are any results to be returned.
-    if (keepLeftUnmatched || keepMatched || keepRightUnmatched) {
-      joinResult = strategy.join(this, right, conditions);
-    }
+    JoinResult joinResult = (keepLeftUnmatched || keepMatched || keepRightUnmatched) ? joinResult = strategy.join(this, right, conditions) : null;
```

Comment:
```suggestion
    JoinResult joinResult = (keepLeftUnmatched || keepMatched || keepRightUnmatched) ? strategy.join(this, right, conditions) : null;
```

Something looks fishy here!

Also what's wrong about the `if`?